### ISSUE TITLE
WebDav

### ISSUE THREAD

**Issue by rijabo** [buttercup/buttercup-mobile#159]
Hi,

Since version 1.8 i android version I cant connect to my WebDav. Gives no network or connection
With the same settings it works with 1.7  but after 1.8 and 1.8.2 now it does not work anymore.
After i reinstall 1.7 it works again.

I waited after version 1.8 ( no connection to webDav)  and installed 1.8.2 now but still no connection with same settings. After reinstal 1.7 works again.

Since i did not find any issues from other users i wanted to ask if this is a know issue.

Thanks in advance.

Regards

**Comment by sallar** [comment]
Hi @rijabo 
I just tried replicating the issue on my test device (OnePlus 5T) and I could not. Can you please share more information about your device, operating system and WebDAV server?

Thank you.

**Comment by rijabo** [comment]
Hi i have a samsung note 9 and webdav is on my nas server
When i install 1.8 or 1.8.2 over the 1.7 version i get attached error
before it works like a charm
When i fresh install 1.8 or 1.8.2 (unistall 1.7) and when input network
adress same as before etc i get network error or no network.

Thanks

**Comment by Oraagon** [comment]
Hello i have the same problem since i updated to 1.8.2
I have a Samsung S8 (Android 9) and use WebDav over http (i know that's not the securest way) on my Synology DS209+

maybe this helps

**Comment by ed-asriyan** [comment]
Hi! I have Huawei Nova 3, Android 9, and the same problem with v1.8.2. I have noticed that issue is reproducing only with http connection, https works well on my device.

**Comment by perry-mitchell** [comment]
Yeah, that’s doable of course. Our Dropbox client also needs an update so both can go together.

**Comment by cropd** [comment]
> Yeah, that’s doable of course. Our Dropbox client also needs an update so both can go together.

That would be great! I just made an iOS build with `"webdav": "^2.9.1"` and I can confirm, it's now connecting to my webdav server.

**Comment by rijabo** [comment]
Im sorry to say i still have te same problem. As i understand it is because i use http://myip:xxxx/webdav and not https . Like i stated in the beginning it worked before. Can it be fixed so http also works? ( im not so expierenced i can setup https ssl on my nas server 😀) thanks in advance.

**Comment by sallar** [comment]
I think that’s not something we can do unfortunately because of this:

  By default, iOS will block any request that's not encrypted using SSL. If you need to fetch from a cleartext URL (one that begins with http) you will first need to add an App Transport Security exception. If you know ahead of time what domains you will need access to, it is more secure to add exceptions only for those domains; if the domains are not known until runtime you can disable ATS completely. Note however that from January 2017, Apple's App Store review will require reasonable justification for disabling ATS. See Apple's documentation for more information.


https://facebook.github.io/react-native/docs/network

**Comment by perry-mitchell** [comment]
The recommendation being, for right now, to use **free** services like LetsEncrypt where possible. I don't personally think it's unacceptable to expect that most servers these days should be accessible using HTTPS - but I completely get that this is a pain. I had no idea RN had this restriction. That being said - please take steps to secure your infrastructure, even if it's just a home service :)

**Comment by casasfernando** [comment]
@perry-mitchell continuing here since https://github.com/buttercup/buttercup-mobile/issues/293 is now closed.

I’ve been doing some research (I’m completely new to mobile development bear that in mind) and found something in the Android development docs (https://developer.android.com/training/articles/security-config) that’s not too complex and may work to allow the app to use trusted CA certs from the system user store.

Basically is about adding the user certificate store to the trusted anchors in the app network security configuration which by default is not included since Android 7.0:

```
<?xml version="1.0" encoding="utf-8"?>
<network-security-config>
<base-config cleartextTrafficPermitted="false">
    <trust-anchors>
        <certificates src="system" />
        <certificates src="user" />
    </trust-anchors>
</base-config>
</network-security-config>
```

Then this new configuration file needs to be included in the app manifest:

```
<?xml version="1.0" encoding="utf-8"?>
<manifest ... >
    <application android:networkSecurityConfig="@xml/network_security_config"
                    ... >
        ...
    </application>
</manifest>
```

I think this is still safer than other solutions since it will still require the user to manually install (like in iOS, Linux, etc) the trusted CA certificate in the device.

What do you think?
If you want to read more all the information is in the Android dev docs link shared above.

Thanks.

**Comment by casasfernando** [comment]
@perry-mitchell did you have the chance to read https://github.com/buttercup/buttercup-mobile/issues/159#issuecomment-1025202284? Any thoughts?

**Comment by casasfernando** [comment]
I just created a PR that fixes https://github.com/buttercup/buttercup-mobile/issues/293 and most likely this issue as well.
I tested it on an Android 11 device and I was able to access the webdav fault without any obvious problems.

Hope it helps.
Thanks.

**Comment by perry-mitchell** [comment]
>I just created a PR that fixes https://github.com/buttercup/buttercup-mobile/issues/293 and most likely this issue as well

I've merged, but I'm currently on holidays. I'll try to release in the next few days..

**Comment by casasfernando** [comment]
> > I just created a PR that fixes #293 and most likely this issue as well
> 
> I've merged, but I'm currently on holidays. I'll try to release in the next few days..
> 
> Of course that PR only addresses the Android side, and not the iOS side..

Thanks for accepting the PR. 😄 

AFAIK iOS didn’t need any fix. I’m using Buttercup on iOS with a private CA certificate for months now without any issues at all. #293 only affected Android.
Yet I must admit I didn’t try self-signed certs in either platform but as long as the user trust the certificate at the OS level it should just work on both.

No rush for the new Android release and enjoy your time off.

Thanks

**Comment by solomon234** [comment]
Bumping this issue, I'm noticing the webdav server I host on my synology has a self signed certificate that typically gets a prompt to trust the server cert. Android Mobile app is unable to connect.

**Comment by perry-mitchell** [comment]
@solomon234 Have you checked [the PR notes](https://github.com/buttercup/buttercup-mobile/pull/305)? This should have been solved.

I'd accept PRs for fixes here, as I won't have time to get to this myself.

**Comment by solomon234** [comment]
Sure thing I'll explore a fix and PR if i find one.
Thanks!

**Comment by dmg-interview** [comment]
I wonder if maybe it could support certificate validation using DANE?

**Comment by tuaris** [comment]
Could certificate validation using DANE be a possibility?

**Comment by perry-mitchell** [comment]
Not sure what that is sorry. But whatever solution is accepted should be at least usable for all android or iOS users, or even better, usable for everyone.

**Comment by solomon234** [comment]
I haven't been able to look into it yet, I assume though that there should at least be a prompt to continue if the CA is trusted.

**Comment by casasfernando** [comment]
> Bumping this issue, I'm noticing the webdav server I host on my synology has a self signed certificate that typically gets a prompt to trust the server cert. Android Mobile app is unable to connect.

Did you try to install your private root CA certificate (the one used to sign your servers certificate) into your Android device as a trusted root CA certificate?
Try that and check if it works. It certainly works for me since the PR above was merged.

**Comment by solomon234** [comment]
good idea, let me look into that and drop an update.
